src/main/php/websocket/Listener.class.php

```diff
@@ -1,13 +1,29 @@
 <?php namespace websocket;
 
-interface Listener {
+abstract class Listener {
```

Comment:
This is a BC break and will mandate a major version release (3.0.0 at the time of writing). This could be solved without this constraint at the cost of creating additional complexity.